tensorflow/core/kernels/conv_ops.cc

```diff
@@ -982,144 +987,268 @@ void LaunchConv2DOp<GPUDevice, T>::operator()(
                                     dtype,                    // tensor datatype
                                     device_id,                // device_id
                                     conv_desc.group_count()};
-  AlgorithmConfig algorithm_config;
 #if TENSORFLOW_USE_ROCM
   // cudnn_use_autotune is applicable only the CUDA flow
   // for ROCm/MIOpen, we need to call GetMIOpenConvolveAlgorithms explicitly
   // if we do not have a cached algorithm_config for this conv_parameters
   cudnn_use_autotune = true;
 #endif
-  if (cudnn_use_autotune &&
-      !AutoTuneConv::GetInstance()->Find(conv_parameters, &algorithm_config)) {
-#if GOOGLE_CUDA
-    std::vector<AlgorithmDesc> algorithms;
-    OP_REQUIRES(
-        ctx,
-        stream->parent()->GetConvolveAlgorithms(
-            conv_parameters.ShouldIncludeWinogradNonfusedAlgo<T>(
-                stream->parent()),
-            &algorithms),
-        errors::Unknown("Failed to get convolution algorithm. This is probably "
-                        "because cuDNN failed to initialize, so try looking to "
-                        "see if a warning log message was printed above."));
-
-    se::TfAllocatorAdapter tf_allocator_adapter(ctx->device()->GetAllocator({}),
-                                                stream);
-    se::RedzoneAllocator rz_allocator(stream, &tf_allocator_adapter,
-                                      se::GpuAsmOpts());
-    se::DeviceMemory<T> output_tensor(
-        WrapRedzoneBestEffort(&rz_allocator, output_ptr));
-
-    std::vector<tensorflow::AutotuneResult> results;
-    for (const auto& profile_algorithm : algorithms) {
-      // TODO(zhengxq): profile each algorithm multiple times to better
-      // accuracy.
-      se::RedzoneAllocator rz_scratch_allocator(
-          stream, &tf_allocator_adapter, se::GpuAsmOpts(),
-          /*memory_limit=*/ConvolveScratchSize);
-      DnnScratchAllocator scratch_allocator(ConvolveScratchSize, ctx);
-      se::ScratchAllocator* allocator_used =
-          !RedzoneCheckDisabled()
-              ? static_cast<se::ScratchAllocator*>(&rz_scratch_allocator)
-              : static_cast<se::ScratchAllocator*>(&scratch_allocator);
-
-      ProfileResult profile_result;
-      auto cudnn_launch_status = stream->ConvolveWithAlgorithm(
-          input_desc, input_ptr, filter_desc, filter_ptr, conv_desc,
-          output_desc, &output_tensor, allocator_used,
-          AlgorithmConfig(profile_algorithm), &profile_result);
-      if (cudnn_launch_status.ok() && profile_result.is_valid()) {
-        results.emplace_back();
-        auto& result = results.back();
-        result.mutable_conv()->set_algorithm(profile_algorithm.algo_id());
-        result.mutable_conv()->set_tensor_ops_enabled(
-            profile_algorithm.tensor_ops_enabled());
 
-        result.set_scratch_bytes(
+  if (CudnnUseFrontend()) {
```

Comment:
Can you try to merge cuDNN frontend case vs the other case? e.g.
```
std::vector<AlgorithmConfig> configs;
if (CudnnUseFrontend()) {
  /* populate configs using execution plans */
} else {
  /* populate configs using algorithms */
}
// Unified code to do autotuning...
for (const auto& config : configs) {
  if (CudnnUseFrontend()) {
    stream->ConvolveWithExecutionPlan(...);
  } else {
    stream->ConvolveWithAlgorithm(...);
  }
}
...
```

Reply:
Done.